dev/PushNotifications/PushNotificationUtility.h

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#pragma once
+#include "pch.h"
+
+const std::wstring ConvertByteArrayToWideString(unsigned int payloadLength, _In_reads_(payloadLength) byte* payload);
+void ProtocolLaunchHelper(unsigned int payloadLength, _In_reads_(payloadLength) byte* payload);
+
+inline HRESULT GetAppUserModelId(wil::unique_cotaskmem_string& appUserModelId)
+{
+    wchar_t appId[APPLICATION_USER_MODEL_ID_MAX_LENGTH] = {};
+    UINT32 appIdSize{ ARRAYSIZE(appId) };
+
+    GetCurrentApplicationUserModelId(&appIdSize, appId);
```

Comment:
Presumably this should check for the one "no package identity" error and return null, then throw otherwise?

Reply:
I thibk this is missing a THROW_IF_FAILED @sharath2727 ?

Reply:
It's OK to have it return null, but the caller has to know that through a comment in the header - "method returns nullptr when the package does not have identity."

Reply:
Sure. I think throwing on the error codes: https://docs.microsoft.com/en-us/windows/win32/api/appmodel/nf-appmodel-getcurrentapplicationusermodelid should take care of this.